doc/voq/voq_hld.md

```diff
@@ -0,0 +1,809 @@
+- max-cores
+- Neighbor Encap index
+
+## 1.7 Scalability requirements
+
+###### Table 2: VOQ scalability
+| VOQ component            | Expected value              |
+|--------------------------|-----------------------------|
+| VOQ Switches             | 128                         |
+| System Ports             | 4K                          |
+| VOQ Neighbors            | 16k                         |
+
+The initial scaling should at least support the requirements of a "large" modular chassis deploying ports based routing interfaces. Virtual interfaces (VLAN based routing interfaces) and directly attached hosts are still supported, but achieving higher scaling for those deployments is not a goal of the initial implementation.
+
+## 1.8 Warm Restart requirements
+At this time warm restart capability is not factored into the design. This shall be revisited as a Phase #2 enhancement
+
+# 2 Modules Design
+### 2.1 System Port Configuration on Sonic Instance
+The system ports are configured on the Line Card. This information is then populated in the "System Port Table" in the "Config DB". On each sonic instance - the "OrchAgent" daemon subscribes to this information and recevies it. 
+
+ ![](../../images/voq_hld/control-card-system-port-config-flow.png)
+
+## 2.2 Config DB
+
+### 2.2.1 DEVICE_METADATA
+ The **existing** DEVICE_METADATA table is enhanced to add new entry to have VOQ related parameters
+ 
+```
+DEVICE_METADATA: {
+   "voqsystem": {
+    "switch_type": {{switch_type}}
+    "switch_id": {{switch_id}}
+    "max_cores" : {{max_cores}}
+   }
+}
+```
+
+VOQ_DB will be specified using database_config.json.j2 and it will generated in control card and linecard differently. Control card will be the server and line card will be running clients only. DBConnectors will use this config to connect correct VOQ_DB.
+
+sonic-buildimage/dockers/docker-database/database_config.json.j2
+
+```
+    "INSTANCES": {
+        "redis":{
+            "hostname" : "{{HOST_IP}}",
+            "port" : 6379,
+            "unix_socket_path" : "/var/run/redis{{NAMESPACE_ID}}/redis.sock",
+            "persistence_for_warm_boot" : "yes"
+        },
+{%- if sonic_asic_platform == "voq" %}
+        "voq-redis":{
+            "hostname" : "{{HOST_IP}}",
+            "port" : 6379,
+            "unix_socket_path" : "/var/run/redis{{NAMESPACE_ID}}/redis.sock",
+            "persistence_for_warm_boot" : "no",
+{%- if sonic_asic_platform_cardtype == "linecard" %}
+            "run_server": "no"
+{%- endif %}
+        }
+{%- endif %}
+    },
+    "DATABASES" : {
+        "APPL_DB" : {
+            "id" : 0,
+            "separator": ":",
+            "instance" : "redis"
+        },
+        ...
+        "VOQ_DB" : {
+            "id" : 8,
+            "separator": ":",
+            "instance" : "voq-redis"
+        },
+```
+        
+
+### 2.2.2 System Port Table
+A **new** table for system port configuration
+
+```
+SYSTEM_PORT:{{system_port_name = PORT.port_name}}
```

Comment:
I think we should name the SYSTEM_PORT like this - 

ASIC_NAME|LOCAL_PORT_NAME

This way, we don't need to keep a separate local name for the interface in this table and all combinations of information is available everywhere. The key for something like SYSTEM_INTERFACE can also follow the same pattern.

Reply:
This is one suggested model out of 2 alternate models. We can have any string describing the slot, asic and port name in any order for the system port name. Only requirement is we should not have ":" in the string. This is mentioned in the note below the alternate model (after OR).

Reply:
Recommendation - 
SYSTEM_PORT|Linecard3.1|Ethernet0
SYSTEM_PORT|Linecard4.0|Ethernet0